pkg/webhook/validator.go

```diff
@@ -1063,6 +1087,332 @@ func (v *Validator) resolvePodSpec(ctx context.Context, ps *corev1.PodSpec, opt
 	resolveEphemeralContainers(ps.EphemeralContainers)
 }
 
+const ResultsAnnotationKey = "policy.sigstore.dev/policy-controller-results"
+
+// AnnotatePod implements duckv1.PodValidator
+func (v *Validator) AnnotatePod(ctx context.Context, p *duckv1.Pod) {
+	// Don't mess with things that are being deleted or already deleted or
+	// status update.
+	if isDeletedOrStatusUpdate(ctx, p.DeletionTimestamp) {
+		return
+	}
+
+	// Attach the spec/metadata for down the line to be attached if it's
+	// required by policy to be included in the PolicyResult.
+	ctx = IncludeSpec(ctx, p.Spec)
+	ctx = IncludeObjectMeta(ctx, p.ObjectMeta)
+
+	imagePullSecrets := make([]string, 0, len(p.Spec.ImagePullSecrets))
+	for _, s := range p.Spec.ImagePullSecrets {
+		imagePullSecrets = append(imagePullSecrets, s.Name)
+	}
+
+	ns := getNamespace(ctx, p.Namespace)
+	opt := k8schain.Options{
+		Namespace:          ns,
+		ServiceAccountName: p.Spec.ServiceAccountName,
+		ImagePullSecrets:   imagePullSecrets,
+	}
+
+	v.annotatePodSpec(ctx, ns, p.Kind, p.APIVersion, &p.ObjectMeta, &p.Spec, opt)
```

Comment:
We could get from the context whether annotate-results- configuration is enabled or not.

Reply:
Sorry, I am not sure what you are referring to. I make the check from the context in one place in line 1206.

Reply:
Oh! This comment was related to this other one https://github.com/sigstore/policy-controller/pull/732#discussion_r1197431379. I believe we could simply get from the context whether our system had enabled the annotations on resources or not.That'd reduce the amount of code for this feature while keeping the behaviour configurable.